### PR TITLE
Ignore freud's generated Cython C++ files in git.

### PR DESCRIPTION
**Pull request by bdice**
## Description
We've previously made the decision to _only_ commit freud's Cython-generated C++ files at release time. To prevent new developers from accidentally adding those files (which happens very frequently), I have added those files to the `.gitignore` and updated the release documentation.

I also updated the garnett compatibility layer since it touches some of the same "requirements" files.

## Motivation and Context
Simplify development and code review process.

## How Has This Been Tested?
`.gitignore` works as expected, builds work on Linux and Windows.

## Types of changes

- [x] Maintenance 🔨

## Checklist:
- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by vyasr** [comment]
We were going to consider just making Cython a hard requirement at some point. Should we consider that for the next minor release? I don't know that "Cython can be hard to install" is really meaningful anymore.

**Comment by bdice** [comment]
@vyasr I can do that instead, it would be an extension of this pull request's existing changes.

**Review by vyasr — APPROVED** [review]
@bdice is something wrong with the way that we're configuring our conda environment for Appveyor? This PR otherwise looks fine to me, and Appveyor failures are usually unrelated to anything else, so I'm fine merging this and addressing that separately if you prefer.

**Comment by bdice** [comment]
There is a bug in Cython affecting parallel cythonization on Windows: https://github.com/cython/cython/issues/3262

I included a workaround for this, so that the Appveyor builds will succeed again.

File: ChangeLog.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ The format is based on
 and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## next
+
+### Added
+- Support for garnett 0.7.
+
+### Changed
+- Cython is no longer an optional dependency. Cythonized `.cpp` files have been removed.
```
Comment (bdice) [inline ChangeLog.md]:
```suggestion
- Cython is now a required dependency (not optional). Cythonized `.cpp` files have been removed.
```

File: freud/locality.pyx [hunk 2]
```diff
@@ -330,9 +330,13 @@ cdef class NeighborQuery:
                 box[[2, 4, 5]] = 0
             system = (box, system.particles.position)
 
-        # garnett compatibility
+        # garnett compatibility (garnett >=0.5)
         elif match_class_path(system, 'garnett.trajectory.Frame'):
-            system = (system.box, system.positions)
+            try:
+                position = system.position
```
Comment (bdice) [inline freud/locality.pyx:337]:
```suggestion
                # garnett >= 0.7
                position = system.position
```

File: freud/locality.pyx [hunk 3]
```diff
@@ -330,9 +330,13 @@ cdef class NeighborQuery:
...
+            except AttributeError:
+                position = system.positions
```
Comment (bdice) [inline freud/locality.pyx:340]:
```suggestion
                # garnett < 0.7
                position = system.positions
```

**Pull request merged**